Library/Homebrew/cli/named_args.rb

```diff
@@ -103,6 +103,34 @@ def to_formulae_to_casks(only: parent&.only_formula_or_cask, method: nil)
                                                 .map(&:freeze).freeze
       end
 
+      # Returns formulae and casks after validating that a tap is present for each of them.
+      def to_formulae_and_casks_with_taps
+        formulae_and_casks_with_taps, formulae_and_casks_without_taps =
+          to_formulae_and_casks.partition do |formula_or_cask|
+            case formula_or_cask
+            when Formula, Cask::Cask
+              formula_or_cask.tap&.installed?
+            else
+              # TODO: Refactor methods so that Sorbet can tell that this is unreachable.
+              raise ArgumentError, "Not a formula or cask: #{formula_or_cask.class}"
+            end
+          end
```

Comment:
```suggestion
          to_formulae_and_casks.partition { |formula_or_cask| formula_or_cask.tap&.installed? }
```

If Sorbet was smarter, it would look like this. I think that things will have to be refactored a bit in the named args class and I'm not sure if that's in scope here.

Reply:
```suggestion
          to_formulae_and_casks.partition { |formula_or_cask| T.cast(formula_or_cask, T.any(Formula, Cask)).tap&.installed? }
```
doesn't work?

Reply:
@carlocab Yep, that worked. Thanks!